### DISCUSSION TITLE
How can I extract the scheme, host and port to use another route in a response?

### DISCUSSION THREAD

**Discussion by FritsHoogland** [tokio-rs/axum#2389] [Q&A]
### Summary

I have a route that generates html that gets sent to the client, which contains a picture on the same server in another route:

The function the route is bound to is below, together with what I have been trying so far.
Currently, I have set scheme, host and port set to fixed values. However, I want it to pick up these from the client request, so that any client GET request that makes it to this route will be able to pick up the image:
```
async fn plot_handler_html(OriginalUri(url): OriginalUri) -> axum::response::Html<String> {
 let scheme = url.scheme_str().unwrap_or("unknown");
 let host = url.host().unwrap_or("unknown");
 let port = if let Some(port) = url.port() { format!(":{}", port) } else { String::from(":unknown") };
 let server_spec = format!("{}://{}{}", scheme, host, port);
 format!(r#"
 format!(r#"
 <!doctype html>
 <html>
  <body>
   <img src="{server_spec}/img">
  </body>
 </html>
 "#).into()
}
```
The problem here is that the response is:
```
➜ curl http://localhost:3000

 <!doctype html>
 <html>
  <body>
   <img src="unknown://unknown:unknown/img">
  </body>
 </html>
```
For which I would expect, and hope to accomplish with your help to be http://localhost:3000/img as image source.

### axum version

0.7.1

**Comment by anarchodin** [comment]
The request doesn't contain that information. In theory, the `Host` header is required now, but that doesn't include the scheme, so even that isn't quite enough.

I assume you actually need this for something other than the given example, because that is trivially the same as using `<img src="/img">`. (Unless you're using `<base>` with a different origin, in which case my sympathy is extremely limited.)
